Yoast/wordpress-seo
# Remove Farsi feature flag

## Context


*

## Summary


This PR can be summarized in the following changelog entry:

* [yoastseo] Removes the Farsi feature flag.
* Completes the readability analysis for Farsi by adding the transition words, sentence beginnings and passive voice assessments, and by adjusting the sentence length assessment to make it more appropriate for the Farsi language.

## Relevant technical choices:

*

## Test instructions

### Test instructions for the acceptance test before the PR gets merged
This PR can be acceptance tested by following these steps:

### To test that the Farsi readability functionality works:

* Set your site to Farsi (فارس).
* Open/create a post and add a text of at least 300 words.
* Make sure that the transition words, passive voice, and consecutive sentences assessments show up.
* Add the word `باری` to the text. Make sure that it's detected by the transition word assessment.
* Add the sentence '.او ربوده شد`. Make sure that it's detected by the passive voice assessment.
* Add the sentences `این زیبا ترین گربه است. این شیرین ترین گربه لاک پشت است. این گربه ای است که همه او را دوست دارند.`. Make sure that the consecutive sentences assessment returns a green bullet.
* Remove all text and paste the following sentence: `.سگ گربه اسب گاو خرگوش همستر خوک مرغ جغد موش خوکچه هندی پروانه تک شاخ خرس اردک جوجه تیغی پشه موش روباه شیر گوزن`
* The sentence length assessment should return a green bullet.

### To test that there is no morphology data or stemmer:
* There is an accompanying Premium PR for these changes: https://github.com/Yoast/wordpress-seo-premium/pull/3440
* Activate Premium (if testing from the branch: use `composer require yoast/wordpress-seo:dev-LINGO-1010-remove-farsi-feature-flag@dev`)
* Check that there is no morphology request by opening a post, opening the console, going to the Networks tab and filtering for 'morphology'. Nothing should be found.
* Set your keyphrase to `cat`.
* Paste the word `cats` somewhere in the text. The keyphrase density assessment should _not_ recognise `cats` as a form of the keyphrase.

### To test that the deprecation works:

* Open the `functions.php` file for your WordPress theme (the location of the file would be something like `plugin-development-docker/wordpress/wp-content/themes/twentytwentyone/functions.php` if testing in Docker, or `Local Sites/test1/app/public/wp-content/themes/twentytwentyone/functions.php` in Local by Flywheel.)
* Enter the following code on top of the file, after the `Functions and definitions` line:

```
$flag = new \Yoast\WP\SEO\Conditionals\Farsi_Support_Conditional();
$flag->is_met();
```
* Activate the Free plugin and click on the WordPress icon in top left corner.
* The admin bar should show one PHP error:
<img width="310" alt="Screenshot 2021-07-09 at 15 35 38" src="https://user-images.githubusercontent.com/32479012/125087553-dd8eb700-e0cc-11eb-9d83-233980cbcc64.png">

* After clicking on it, the following warning should show (with the name of the relevant feature flag):
<img width="1093" alt="Screenshot 2021-07-09 at 15 37 28" src="https://user-images.githubusercontent.com/32479012/125086103-981dba00-e0cb-11eb-8a14-715b1826040b.png">

* Remove the line of code after testing


### Test instructions for QA when the code is in the RC




QA can test this PR by following these steps:

*

## Impact check

This PR affects the following parts of the plugin, which may require extra testing:

*

## UI changes


## Documentation


## Quality assurance


Fixes https://yoast.atlassian.net/browse/LINGO-1010
